### PR TITLE
allow-file-access

### PR DESCRIPTION
**Pull request by rojters**

### CODE REVIEW

File: typo/typo.js [hunk 1]
```diff
@@ -217,7 +222,7 @@ Typo.prototype = {
 			if (async) {
 				promise = new Promise(function(resolve, reject) {
 					req.onload = function() {
-						if (req.status === 200) {
+						if (req.status === 200 || (isFile && req.status === 0)) {
```
Comment (EvilOatmeal) [inline typo/typo.js]: Please add a comment that explains why you're checking that status === 0, or extract that expression into a variable with a nice name.

File: typo/typo.js [hunk 2]
```diff
@@ -210,6 +210,11 @@ Typo.prototype = {
 		if (!charset) charset = "utf8";
 		
 		if (typeof XMLHttpRequest !== 'undefined') {
+			var isFile = false;
+			if (location.protocol === "file:") {
+				isFile = true;
+			}
+
```
Comment (EvilOatmeal) [inline typo/typo.js]: Reduce this to: `var isFile = location.protocol === "file:";`

Comment (rojters) [inline typo/typo.js]: Done.

File: typo/typo.js (hunk 1 above)
Comment (rojters) [inline typo/typo.js]: Done.

**Pull request merged**